pkg/api/pod/util.go

```diff
@@ -770,11 +770,11 @@ func podLifecycleSleepActionZeroValueInUse(podSpec *api.PodSpec) bool {
 		if c.Lifecycle == nil {
 			return true
 		}
-		if c.Lifecycle.PreStop != nil && c.Lifecycle.PreStop.Sleep != nil && c.Lifecycle.PreStop.Sleep.Seconds == 0 {
+		if lc := c.Lifecycle.PreStop; lc != nil && lc.Sleep != nil && lc.Sleep.Seconds == 0 {
 			inUse = true
 			return false
 		}
-		if c.Lifecycle.PostStart != nil && c.Lifecycle.PostStart.Sleep != nil && c.Lifecycle.PreStop.Sleep.Seconds == 0 {
+		if lc := c.Lifecycle.PostStart; lc != nil && lc.Sleep != nil && lc.Sleep.Seconds == 0 {
```

Comment:
```suggestion
		if c.Lifecycle.PostStart != nil && c.Lifecycle.PostStart.Sleep != nil && c.Lifecycle.PostStart.Sleep.Seconds == 0 {
```

Reply:
This factoring avoids the whole class of bug. I think we would be wise to embrace this pattern.

Reply:
for sure, just not in the change we're backporting

Reply:
I can take this up as part of [KEP 4818](https://github.com/kubernetes/enhancements/issues/4818) and refactor all the occurrences before code freeze. I can see the same pattern for PostStart and PreStop in a bunch of places from both KEP 4818 and [KEP 3960](https://github.com/kubernetes/enhancements/issues/3960)

Reply:
Hey, just saw this conversation. Let me know what the decision is here and I'd be happy to update the PR accordingly. I know @sreeram-venkitesh is trying to get this merged to unblock KEP4818 so I'd hate to accidentally block that. Also feel free to update the PR directly if you prefer - whatever works

Reply:
I think we're looking to make this specific PR as tiny and targeted as possible to fix the exact bug, so we can backport it to the release branch as fast as possible. Following up with another PR in master that fixes the pattern the way you did here sounds good to me.

Reply:
that's fair. Removed unrelated changes